### PR TITLE
Fixing build errors on ARM NEON.

### PR DESCRIPTION
**Pull request by zdydek**
I was trying to build this for an ARM platform, but ran into some problems.  Not sure if my changes are the best way to resolve them, but the code does compile now.

### CODE REVIEW

**Comment by ebassi** [comment]
Thanks! I'm setting up an ARM device to test out the changes (and maybe have some sort of automated regression testing), but in principle they look good to me…

File: src/graphene-matrix.c [hunk 1]
```diff
@@ -746,7 +746,25 @@ graphene_matrix_get_value (const graphene_matrix_t *m,
       return 0.f;
     }
 
-  return graphene_simd4f_get (r, col);
+  switch (col)
```
Comment (ebassi) [inline src/graphene-matrix.c]: Since this has nothing to do with ARM NEON issues, could you split this chunk into its own separate PR?

File: src/graphene-simd4f.c [hunk 2]
```diff
@@ -216,7 +216,19 @@ float
 (graphene_simd4f_get) (const graphene_simd4f_t s,
                        unsigned int            i)
 {
-  return graphene_simd4f_get (s, i);
+  switch (i)
```
Comment (ebassi) [inline src/graphene-simd4f.c]: Same as above, this should go into a separate PR.

File: src/graphene-matrix.c (hunk 1 above)
Comment (zdydek) [inline src/graphene-matrix.c]: It does have something to do with ARM NEON, though it is a bit obfuscated.  The underlying ARM fast type that graphene_simd4f_get calls into (I forget which one) requires its inputs to be compile time constants.  Passing the "col" argument in was causing the build to fail.

If you still want me to break out into a separate commit I can do that.

File: src/graphene-simd4f.c (hunk 2 above)
Comment (zdydek) [inline src/graphene-simd4f.c]: Ditto.

File: src/graphene-matrix.c (hunk 1 above)
Comment (ebassi) [inline src/graphene-matrix.c]: Ah, that would have been clearer in a separate commit indeed. ;-)

Yes, if you could split this change in a separate commit and note why it's necessary in the commit log it would be very much welcome — especially because it'll avoid me getting rid of it in the future when I've forgotten about this issue.

**Comment by james-huang** [comment]
hi guys, I've independently some across several of the same issues trying to build arm.  
[commit](https://github.com/james-huang/graphene/commit/11fd3145fcf745109cee514d3d0483be1ce5a36e)

although instead of vreinterpretq_u8_u32 i went with vcgeq_u8.
is vreinterpretq_u8_u32 android specific? 

[ARM Compiler toolchain Compiler Reference] (http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0491f/BABGDDDH.html)

**Comment by ebassi** [comment]
I guess we can close this.

**Comment by ebassi** [comment]
@james-huang I've commented on #65. Looking for `vreinterpretq_u8_u32`, I seem to find it in various toolchains for the NEON set in the arm-eabi, so I'm not sure it's Android specific.
